### DISCUSSION TITLE
Uninformative marker genes

### DISCUSSION THREAD

**Discussion by rbutleriii** [satijalab/seurat#5018] [Q&A] [answered]
I have read about the concept of a cluster with "uninformative gene markers", but don't see it really discussed in the tutorials. And in my data, if I run a cursory clustering:
```
  sc = subset(sc, subset=nFeature_RNA > 200 & nCount_RNA > 500 & percent.mt < 0.1)
  sc = SCTransform(sc, vars.to.regress="percent.mt")
  sc = RunPCA(sc, npcs=10)
  sc = RunUMAP(sc, dims=1:10)
  sc = FindNeighbors(sc, dims=1:10, verbose=F, reduction="pca")
  sc = FindClusters(sc, verbose=F)
  # Identify marker genes
  DefaultAssay(sc) = "RNA"
  markers = data.table(FindAllMarkers(sc, only.pos=T, min.pct=0.25, 
                                      logfc.threshold=0.25))
  marker = markers[order(p_val_adj, -avg_log2FC)][, head(.SD, 20), by=c("cluster")]
  marker[, rank := frank(.(p_val_adj, -avg_log2FC), ties.method="dense"), by=c("cluster")]
  top20 = dcast(marker, rank ~ cluster, value.var="gene")
  fwrite(top20, file=paste(today, nameset, meta$sample_id[i], "top20genes.csv", 
                           sep="_"))
``` 
I get a one or two clusters that have only 5-10 marker genes (this is consistent across several independently sequenced samples), that are mito or rpl genes, for example: mt-Co3, Pcsk1n, Cox4i1, Rplp1, Atp6v0c, Ubb, Rpl13, Rpl19, Gm20404. More odd as this is snRNA-seq, so mito counts should be largely contamination. Are these artifacts? DoubletFinder suggests removing uninformative clusters and re-clustering before proceeding, so are these the infamous bunch?
![image](https://user-images.githubusercontent.com/6644896/131453238-63e0451f-2dce-4eb5-8f50-fecfa7f043dd.png)
[20210827_PS19_C31_B334_340_top20genes.csv](https://github.com/satijalab/seurat/files/7081722/20210827_PS19_C31_B334_340_top20genes.csv)

**Comment by saketkc** [comment, marked as answer]
This can sometimes happen because of having higher pearson residuals for these genes which could be controlled by clipping them to a smaller range (default is [-sqrt(ncells/30), sqrt(ncells/30)]). If you can email the object at schoudhary@nygenome.org, I might be able to take a close look.

**Reply by rbutleriii** [reply]
Thanks @saketkc , solved by remembering to `NormalizeData()` on the RNA assay prior to `FindAllMarkers()`.
